internal/update/realtime/realtime.go

```diff
@@ -588,3 +605,171 @@ func convertOptionalTime(in *time.Time) pgtype.Timestamptz {
 		Time:  *in,
 	}
 }
+
+func updateVehicles(ctx context.Context, updateCtx common.UpdateContext, vehicles []gtfs.Vehicle) error {
+	onlyProcessFullEntities := updateCtx.FeedConfig != nil &&
+		updateCtx.FeedConfig.GetGtfsRealtimeOptions().GetOnlyProcessFullEntities()
+	var validVehicleEntities []gtfs.Vehicle
+	for _, vehicle := range vehicles {
+		// Only insert vehicles that are in the feed.
+		if onlyProcessFullEntities && !vehicle.IsEntityInMessage {
+			continue
+		}
+
+		// TODO: We can insert a vehicle with no ID if it has an associated trip.
+		// For now, we just skip it.
+		if vehicle.GetID().ID == nil || *vehicle.GetID().ID == "" {
+			updateCtx.Logger.DebugCtx(ctx, "Vehicle has no ID or empty ID")
+			continue
+		}
+
+		validVehicleEntities = append(validVehicleEntities, vehicle)
+	}
+
+	if len(validVehicleEntities) == 0 {
+		return updateCtx.Querier.ClearVehicleFeed(ctx, updateCtx.FeedPk)
+	}
```

Comment:
I think it may be better to re-use the stale vehicles query here? Just passing in an empty set of active vehicle IDs.

Reply:
Makes sense, should be fixed in latest revision!